Support Windows ARM64 (`win-arm64`)

weiji14 opened the pull request:
GitHub added Windows ARM64 runners yesterday at https://github.blog/changelog/2025-04-14-windows-arm64-hosted-runners-now-available-in-public-preview/, and `mamba` supports win-arm64 since https://github.com/mamba-org/mamba/pull/2745.

Haven't tested, but should hopefully fix `Error: Unsupported platform: win32-arm64` seen at https://github.com/GenericMappingTools/pygmt/pull/3911#issuecomment-2803365084

weiji14 committed:
Support win-arm64

weiji14 committed:
Support win-arm64

pavelzw commented on src/util.ts:
```diff
@@ -27,7 +27,8 @@ export const getCondaArch = () => {
     'linux-ppc64': 'linux-ppc64le',
-    'win32-x64': 'win-64'
+    'win32-x64': 'win-64',
+    'win32-arm64': 'win-arm64'
```
Could you add a unit test in .github/workflows/test.yml as well?

weiji14 replied:
Added `windows-11-arm` (and also `ubuntu-24.04-arm`) runners to the matrix build for the "bash" job at commit 9d33c874eefd6a01ba6433280a50c391c6c44dd0, hopefully that's ok?

weiji14 committed:
Test on ARM runners

weiji14 committed:
Test on ARM runners

pavelzw commented:
You still need to run `pnpm install` and `pnpm build`

weiji14 committed:
pnpm install && pnpm build

weiji14 committed:
pnpm install && pnpm build

pavelzw commented on src/util.ts:
```diff
@@ -27,7 +27,8 @@ export const getCondaArch = () => {
     'linux-ppc64': 'linux-ppc64le',
-    'win32-x64': 'win-64'
+    'win32-x64': 'win-64',
+    'win32-arm64': 'win-arm64'
```
as long as https://github.com/mamba-org/micromamba-releases/releases/tag/2.1.0-0 doesn't have binaries for win-arm64, this won't work

weiji14 replied:
Ah yes, and I guess we'll need to create `win-arm64` builds on https://github.com/conda-forge/micromamba-feedstock first, xref https://github.com/conda-forge/conda-forge.github.io/issues/1940.

weiji14 replied:
> as long as https://github.com/mamba-org/micromamba-releases/releases/tag/2.1.0-0 doesn't have binaries for win-arm64, this won't work

Issue tracked at https://github.com/mamba-org/micromamba-releases/issues/91

jjerphan committed:
Bump python to 3.14 to have it installable for win-arm64

Signed-off-by: Julien Jerphanion <git@jjerphan.xyz>

jjerphan committed:
Remove codecov

Not installable for win-arm64 because brotli is
not available for this platform

Signed-off-by: Julien Jerphanion <git@jjerphan.xyz>

JohanMabille reviewed (approved):
Last micromamba release has been edited to include the build for win-arm64, and https://github.com/mamba-org/micromamba-releases/issues/91 has been solved, so I think we can get this in.

JohanMabille committed:
Merge branch 'main' into patch-1
